tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/UsageReport/WritePackageUsageData.cs

```diff
@@ -180,28 +180,37 @@ public override bool Execute()
                 assetFiles,
                 assetFile =>
                 {
-                    var properties = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                    JObject jObj;
 
                     using (var file = File.OpenRead(Path.Combine(RootDir, assetFile)))
                     using (var reader = new StreamReader(file))
                     using (var jsonReader = new JsonTextReader(reader))
                     {
-                        while (jsonReader.Read())
-                        {
-                            if (jsonReader.TokenType == JsonToken.PropertyName &&
-                                jsonReader.Value is string value)
-                            {
-                                properties.Add(value);
-                            }
-                        }
+                        jObj = (JObject)JToken.ReadFrom(jsonReader);
```

Comment:
A reason to do it the earlier way was performance. I don't really think this will be different enough to matter, but once CI is done I think before/after should be compared.

Reply:
Spotcheck of a number of jobs before and after:

Job | Before | After | Diff
-- | -- | -- | --
Ubuntu16.04_Release_prtest | 00:01.7 | 00:02.3 | 00:00.6
CentOS7.1_Release_prtest | 00:01.6 | 00:02.6 | 00:01.0
Debian8.4_Release_prtest | 00:01.5 | 00:02.0 | 00:00.5
RHEL7.2_Release_prtest | 00:01.8 | 00:02.3 | 00:00.5
Ubuntu16.04_Release_prtest | 00:01.7 | 00:02.3 | 00:00.6
  |   | Avg Diff: | 00:00.6

It doesn't seem like this is different enough to be an issue.

Reply:
Thanks for checking, yeah, seems fine.